config_test.go

```diff
@@ -74,6 +89,7 @@ func TestConfigStringToClusterConfig(t *testing.T) {
 		{info: "invalid key", configString: "?foo=bar", err: fmt.Errorf("invalid key: foo")},
 
 		{info: "empty", configString: "", clusterConfig: NewClusterConfig()},
+		{info: "empty", configString: sslConfigString, clusterConfig: sslClusterConfig},
```

Comment:
Need more tests, including empty data.

Reply:
as I added some tests, I found there's a bug https://github.com/MichaelS11/go-cql-driver/blob/60b83bcdb2ef5c698b6481351b2508f3c1734a93/config.go#L96

1. we actually shouldn't be doing `strings.TrimSpace` on both strings
1. we should `url.QueryUnescape` the value, esp since `password=` could be a string with special characters

``` go
key, rawValue := strings.TrimSpace(settingSplit[0]), settingSplit[1]
value, err := url.QueryUnescape(rawValue)
if err != nil {
	return nil, fmt.Errorf("failed for: %v = %v", key, rawValue)
}
```

should I add this into this PR too?

Reply:
7d8f510